pkg/shares/non_interactive_defaults.go

```diff
@@ -0,0 +1,94 @@
+package shares
+
+// FitsInSquare uses the non interactive default rules to see if messages of
+// some lengths will fit in a square of size origSquareSize starting at share
+// index cursor. See non-interactive default rules
+// https://github.com/celestiaorg/celestia-specs/blob/master/src/rationale/message_block_layout.md#non-interactive-default-rules
+func FitsInSquare(cursor, origSquareSize int, msgShareLens ...int) (bool, int) {
+	// if there are 0 messages and the cursor already fits inside the square,
+	// then we already know that everything fits in the square.
+	if len(msgShareLens) == 0 && cursor/origSquareSize <= origSquareSize {
+		return true, 0
+	}
+	firstMsgLen := 1
+	if len(msgShareLens) > 0 {
+		firstMsgLen = msgShareLens[0]
+	}
+	// here we account for padding between the contiguous and message shares
+	cursor, _ = NextAlignedPowerOfTwo(cursor, firstMsgLen, origSquareSize)
+	sharesUsed, _ := MsgSharesUsedNIDefaults(cursor, origSquareSize, msgShareLens...)
+	return cursor+sharesUsed <= origSquareSize*origSquareSize, sharesUsed
+}
+
+// MsgSharesUsedNIDefaults calculates the number of shares used by a given set
+// of messages share lengths. It follows the non-interactive default rules and
+// assumes that each msg length in msgShareLens
```

Comment:
This function does not rely on that assumption, otherwise it would not be able to handles messges that are larger than a single row. It definitely looks like we have to complete those docs, not exactly sure what I was going for originally 91c0d72174b6d7cbcb0233cb51b9f30aeab7faab